internal/incident/incident.go

```diff
@@ -60,351 +74,469 @@ func (i *Incident) HasManager() bool {
+			Time:                      types.UnixMilli(time.Now()),
+			Type:                      Notified,
+			CausedByIncidentHistoryID: i.causedByHistoryID,
 		}
 
-		recipients := make(map[recipient.Key]*RecipientState)
-		for _, contact := range contacts {
-			recipients[contact.Key] = &RecipientState{Role: contact.Role}
-		}
+		for chType := range channels {
+			i.logger.Infof("[%s %s] notify %q via %q", i.Object.DisplayName(), i.String(), contact.FullName, chType)
+
+			hr.ChannelType = utils.ToDBString(chType)
+
+			_, err := i.AddHistory(hr, false)
+			if err != nil {
+				i.logger.Errorln(err)
+			}
+
+			chConf := i.runtimeConfig.Channels[chType]
+			if chConf == nil {
+				i.logger.Errorf("could not find config for channel type %q", chType)
+				continue
+			}
 
-		currentIncident.Recipients = recipients
+			plugin, err := chConf.GetPlugin()
+			if err != nil {
+				i.logger.Errorw("couldn't initialize channel", zap.String("type", chType), zap.Error(err))
+				continue
+			}
+
+			err = plugin.Send(contact, i, ev, i.configFile.Icingaweb2URL)
+			if err != nil {
+				i.logger.Errorw("failed to send via channel", zap.String("type", chType), zap.Error(err))
+				continue
+			}
+		}
 	}
 
-	return currentIncident, created, nil
+	return nil
 }
 
-func RemoveCurrent(obj *object.Object, hr *HistoryRow) error {
-	currentIncidentsMu.Lock()
-	defer currentIncidentsMu.Unlock()
+// processAcknowledgmentEvent processes the given ack event.
+// Promotes the ack author to incident.RoleManager if it's not already the case and generates a history entry.
+// Returns error on database failure.
+func (i *Incident) processAcknowledgementEvent(ev event.Event) error {
+	contact := i.runtimeConfig.GetContact(ev.Username)
+	if contact == nil {
+		return fmt.Errorf("unknown acknowledgment author %q", ev.Username)
+	}
+
+	recipientKey := recipient.ToKey(contact)
+	state := i.Recipients[recipientKey]
+	oldRole := RoleNone
+	newRole := RoleManager
+	if state != nil {
+		oldRole = state.Role
+
+		if oldRole == RoleManager {
+			// The user is already a manager
+			return nil
+		}
+	} else {
+		i.Recipients[recipientKey] = &RecipientState{Role: newRole}
+	}
 
-	currentIncident := currentIncidents[obj]
+	i.logger.Infof("[%s %s] contact %q role changed from %s to %s", i.Object.DisplayName(), i.String(), contact.String(), oldRole.String(), newRole.String())
 
-	if currentIncident != nil {
-		delete(currentIncidents, obj)
+	hr := &HistoryRow{
+		Key:              recipientKey,
+		EventID:          utils.ToDBInt(ev.ID),
+		Type:             RecipientRoleChanged,
+		Time:             types.UnixMilli(time.Now()),
+		NewRecipientRole: newRole,
+		OldRecipientRole: oldRole,
+		Message:          utils.ToDBString(ev.Message),
 	}
 
-	incidentRow := &IncidentRow{ID: currentIncident.incidentRowID, RecoveredAt: types.UnixMilli(currentIncident.RecoveredAt)}
-	_, err := currentIncident.db.NamedExec(`UPDATE "incident" SET "recovered_at" = :recovered_at WHERE id = :id`, incidentRow)
```

Comment:
Looks like this update got lost while refactoring.

When updating this PR, please keep the commits up to (including) c21dac071f257ae79dacba0a656671a95a129a59 as is as I've already looked over those.